include/cpr/threadpool.h

```diff
@@ -67,7 +67,7 @@ class ThreadPool {
         if (status == STOP) {
             Start();
         }
-        if (idle_thread_num <= 0 && cur_thread_num < max_thread_num) {
+        if (idle_thread_num == 0 && cur_thread_num < max_thread_num) {
```

Comment:
I prefer range based checks rather than equality.
This has the simple reason that in the case we get a bit flip or something like it, we are still "safe" and in my eyes it's clearer to read.

Reply:
`idle_thread_num` is a size_t type variable, when it is 0, executing "--idle_thread_num;", it will becomes a super large value instead of a negative number.
If you want to be more safe, is it better to define `idle_thread_num` as a signed number?

Reply:
Jup, I'm aware of that. It's just a general thing for me :)

Reply:
> Jup, I'm aware of that. It's just a general thing for me :)

Sorry for that ~ I think you are right